webservice/models/webservice_consumer_mixin.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 Foodles (http://www.foodles.co).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class WebserviceConsumerMixin(models.AbstractModel):
+
+    _name = "webservice.consumer.mixin"
+    _description = "Add fields to save web service responses"
+
+    ws_response_status_code = fields.Integer(
+        "Status code", help="Web service response HTTP code"
+    )
+    ws_response_content = fields.Binary(
+        "Response", attachment=True, copy=False, help="Web service response content"
+    )
+    ws_response_content_filename = fields.Char(
+        compute="_compute_ws_response_content_filename"
+    )
+
+    def _compute_ws_response_content_filename(self):
+        for rec in self:
+            rec.ws_response_content_filename = (
+                "response_%s.json" % (str(rec.ws_response_status_code),)
```

Comment:
I wanted to reuse efficiently all the business rules implemented here https://github.com/OCA/edi/blob/14.0/edi_oca/models/edi_exchange_type.py#L213

Reply:
at the moment this is a computed field not stored :thinking: I could add a response_at datetime field :thinking:

Reply:
An alternative would be to relate the ir.attachment and read the write date from there...
No strong opinion on this but I think it would be nice to know when the response was recorded.

Reply:
indeed it's a nice to have !

Reply:
done by https://github.com/OCA/web-api/pull/22/commits/b5f821546450888863073d38cc74f02fbffbcab4